gst/nnstreamer/tensor_query/tensor_query_common.c

```diff
@@ -0,0 +1,189 @@
+/* SPDX-License-Identifier: LGPL-2.1-only */
+/**
+ * Copyright (C) 2021 Gichan Jang <gichan2.jang@samsung.com>
+ *
+ * @file   tensor_query_common.c
+ * @date   09 July 2021
+ * @brief  Utility functions for tensor query
+ * @see    https://github.com/nnstreamer/nnstreamer
+ * @author Gichan Jang <gichan2.jang@samsung.com>
+ * @bug    No known bugs except for NYI items
+ * @see    gsttcpserversink, gsttcpserversrc
+ */
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include "nnstreamer_log.h"
+#include "tensor_query_common.h"
+
+/** @todo change max read size or need payload */
+#define MAX_READ_SIZE 4096
+
+/**
+ * @brief Create requested socket.
+ * @param[in] hostname the hostname.
+ * @param[in] port a port number.
+ * @param[in] cancellable (nullable) a #GCancellable, or %NULL.
+ * @return Newly created socket or NULL on error.
+ * @note Caller is responsible for unreferring the returned object with g_object_unref().
+ */
+GSocket *
+gst_tensor_query_socket_new (const gchar * hostname, guint16 port,
+    GCancellable * cancellable)
+{
+  GSocket *socket;
+  GError *err = NULL;
+  GInetAddress *addr;
+  GSocketAddress *saddr;
+
+  /* look up name if we need to */
+  addr = g_inet_address_new_from_string (hostname);
+  if (!addr) {
+    GList *results;
+    GResolver *resolver;
+
+    resolver = g_resolver_get_default ();
+    results = g_resolver_lookup_by_name (resolver, hostname, cancellable, &err);
+
+
+    if (!results) {
+      if (g_error_matches (err, G_IO_ERROR, G_IO_ERROR_CANCELLED)) {
+        nns_logd ("gst_tensor_query_socket_new: Cancelled name resolval");
+      } else {
+        nns_loge ("Failed to resolve host '%s': %s", hostname, err->message);
+      }
+      g_clear_error (&err);
+      g_object_unref (resolver);
+      return NULL;
+    }
+
+    addr = G_INET_ADDRESS (g_object_ref (results->data));
```

Comment:
```results``` looks a Glist of GInetAddress. Then, is it guaranteed that there is only one element in this GList?

Reply:
Right. results have more than one address.
But I think checking all addresses is over-engineering at this stage, as this route is also a defense code.
I marked it as @todo and make up for it next time.